### ISSUE TITLE
The jenkins_slave_ssh_pubkey does NOT update if the id_rsa.pub changes on disk

### ISSUE THREAD

**Issue by halfshellhero** [rackspace-cookbooks/jenkinsstack#23]
Initial build works fine, but if /var/lib/jenkins is copied over from another server for migration purposes and the keys inside /var/lib/jenkins/.ssh/ are overwritten, the jenkins_slave_ssh_pubkey is not updated leading to failure to converge.

Turning security off does not resolve this issue as the new key will not be updated in the node attributes.

Here is the stacktrace generated when the keys change on disk:

Mixlib::ShellOut::ShellCommandFailed: jenkins_user[chef](jenkinsstack::master line 27) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '255'
---- Begin output of java -jar /var/chef/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140819-12555-1202b3w ----
STDOUT: 
STDERR: Authentication failed. No private key accepted.
---- End output of java -jar /var/chef/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140819-12555-1202b3w ----
Ran java -jar /var/chef/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140819-12555-1202b3w returned 255
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/mixlib-shellout-1.4.0/lib/mixlib/shellout.rb:257:in `invalid!'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/mixlib-shellout-1.4.0/lib/mixlib/shellout.rb:244:in`error!'
/var/chef/cache/cookbooks/jenkins/libraries/_executor.rb:79:in `execute!'
/var/chef/cache/cookbooks/jenkins/libraries/_executor.rb:109:in`groovy!'
/var/chef/cache/cookbooks/jenkins/libraries/user.rb:96:in `block (2 levels) in <class:JenkinsUser>'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/mixin/why_run.rb:52:in`call'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/mixin/why_run.rb:52:in `add_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/provider.rb:156:in`converge_by'
/var/chef/cache/cookbooks/jenkins/libraries/user.rb:95:in `block in <class:JenkinsUser>'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/provider/lwrp_base.rb:138:in`instance_eval'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/provider/lwrp_base.rb:138:in `block in action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/provider.rb:121:in`run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource.rb:648:in `run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/runner.rb:49:in`run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/runner.rb:81:in`each'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/runner.rb:81:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection.rb:98:in`block in execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection/stepable_iterator.rb:116:in`call_iterator_block'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection/stepable_iterator.rb:104:in`iterate'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/resource_collection.rb:96:in`execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/runner.rb:80:in `converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/client.rb:345:in`converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/client.rb:431:in `do_run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/client.rb:213:in`block in run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/client.rb:207:in `fork'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/client.rb:207:in`run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/application.rb:236:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/application/client.rb:338:in`block in run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/application/client.rb:327:in `loop'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/application/client.rb:327:in`run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/lib/chef/application.rb:55:in `run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.14.6/bin/chef-client:26:in`<top (required)>'
/usr/bin/chef-client:23:in `load'
/usr/bin/chef-client:23:in`<main>'

I apologize, I lost the chef-client output before creating this issue, but the problem is reproducible by simply updating the ssh keys on disk and attempting a chef-client run.  It would be nice if this cookbook anticipated potential migration of /var/lib/jenkins from another server.

**Comment by bobross419** [comment]
```
Recipe: jenkinsstack::_prep_keys       
  * chef_gem[sshkey] action install[2014-09-03T17:23:11+00:00] INFO: Processing chef_gem[sshkey] action install (jenkinsstack::_prep_keys line 19)       
 (up to date)       
  * ruby_block[prepare_keys] action nothing[2014-09-03T17:23:11+00:00] INFO: Processing ruby_block[prepare_keys] action nothing (jenkinsstack::_prep_keys line 28)       
 (skipped due to action :nothing)       
Recipe: jenkinsstack::master       
  * jenkins_user[chef] action create[2014-09-03T17:23:11+00:00] INFO: Processing jenkins_user[chef] action create (jenkinsstack::master line 27)       
Recipe: <Dynamically Defined Resource>       
  * remote_file[/tmp/kitchen/cache/jenkins-cli.jar] action create[2014-09-03T17:23:11+00:00] INFO: Processing remote_file[/tmp/kitchen/cache/jenkins-cli.jar] action create (dynamically defined)       
 (up to date)       
  * remote_file[/tmp/kitchen/cache/update-center.json] action create_if_missing[2014-09-03T17:23:11+00:00] INFO: Processing remote_file[/tmp/kitchen/cache/update-center.json] action create_if_missing (dynamically defined)       
 (up to date)       

================================================================================       
Error executing action `create` on resource 'jenkins_user[chef]'       
================================================================================       


Mixlib::ShellOut::ShellCommandFailed       
------------------------------------       
Expected process to exit with [0], but received '255'       
---- Begin output of java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv ----       
       STDOUT: 
       STDERR: Authentication failed. No private key accepted.
       ---- End output of java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv ----
       Ran java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv returned 255


       Cookbook Trace:
       ---------------
       /tmp/kitchen/cache/cookbooks/jenkins/libraries/_executor.rb:79:in `execute!'
       /tmp/kitchen/cache/cookbooks/jenkins/libraries/_executor.rb:109:in `groovy!'
       /tmp/kitchen/cache/cookbooks/jenkins/libraries/user.rb:96:in `block (2 levels) in <class:JenkinsUser>'
       /tmp/kitchen/cache/cookbooks/jenkins/libraries/user.rb:95:in `block in <class:JenkinsUser>'


       Resource Declaration:

       ---------------------
       # In /tmp/kitchen/cache/cookbooks/jenkinsstack/recipes/master.rb

        27: jenkins_user 'chef' do
        28:   public_keys [s_public_key]
        29: end
        30: 



       Compiled Resource:
       ------------------

       # Declared in /tmp/kitchen/cache/cookbooks/jenkinsstack/recipes/master.rb:27:in `from_file'

       jenkins_user("chef") do
         action :create
         retries 0
         retry_delay 2
         guard_interpreter :default
         cookbook_name "jenkinsstack"
         recipe_name "master"
         public_keys ["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDhEDhukTBldlQMnnURu9JfsduqA5q0bPSn6xAGgAS5CqjpgKzWSfKnXf2WcuUmM0pKUvU5PhEU37zzPNTP/rwrUaX3UMglMY8NiZC3gS3uzCI2DI3n2feJexbIeO3xMeWQonFiaqqVVBgpGonU38S59uDb3rnk6s8eHwZy/f/5XHrk4DrPS2UPbrrqInfHn5jDMN7wJZY99hnCIm3Tk7bBPbxIs3XyxzlG4RCkpgAwKrDWdOV8C8OJ49F12kM6PqhQEEbvxMSZHI/ftxr3PObYelz9BOFAQXjHMC6R0CvFkHe2UVmVYR8rMZlJ1+lS0bX9l9LqzEwmnvyZDJXePiQ3"]
         id "chef"
       end



[2014-09-03T17:23:16+00:00] INFO: Running queued delayed notifications before re-raising exception       

Running handlers:       
[2014-09-03T17:23:16+00:00] ERROR: Running exception handlers       
Running handlers complete       

[2014-09-03T17:23:16+00:00] ERROR: Exception handlers complete       
[2014-09-03T17:23:16+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out       
Chef Client failed. 3 resources updated in 31.473193102 seconds       
[2014-09-03T17:23:16+00:00] ERROR: jenkins_user[chef] (jenkinsstack::master line 27) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '255'       
---- Begin output of java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv ----       
STDOUT:        
STDERR: Authentication failed. No private key accepted.       
---- End output of java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv ----       
Ran java -jar /tmp/kitchen/cache/jenkins-cli.jar -s http://localhost:8080 -i /var/lib/jenkins/.ssh/id_rsa groovy /tmp/groovy20140903-10074-85j5uv returned 255       
[2014-09-03T17:23:16+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)       
>>>>>> Converge failed on instance <default-ubuntu-1404>.
>>>>>> Please see .kitchen/logs/default-ubuntu-1404.log for more details
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: SSH exited (1) for command: [sudo -E chef-client -z --config /tmp/kitchen/client.rb --log_level info --json-attributes /tmp/kitchen/dna.json]
>>>>>> ----------------------
```

**Comment by martinb3** [comment]
Fixed in #24.
